### DISCUSSION TITLE
How to handle calculated field which depends on related data?

### DISCUSSION THREAD

**Discussion by michaltaberski** [prisma/prisma#19984] [Q&A] [answered]
Hey,

I am struggling with a proper implementation / architecture of following issue:

```
Product -- HAS_MANY --> Photos
```

I would like always product to have a value of url from first photo. eg. `productThumbnailUrl`. I can always query photos on every venue request, but I think it should be doable with `prisma.$extends` ? Or there should be a different technique doing that? Perhaps there is a hook mechanism on photo creation/deletion that updates `productThumbnailUrl` field (ofc. field has to be added in that case). What's the best solution here? Thanks!

**Comment by nurul3101** [comment, marked as answer]
Hello @michaltaberski 👋 

One approach I can think of is Eager Loading, Eager loading is an approach where you retrieve the product along with its related photos in a single query. This reduces the number of database queries but might retrieve more data than you need. You can use  Prisma to facilitate eager loading. You could use the [include](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include) option to retrieve the photos along with the product. This way, you can access the first photo's URL without an additional query.

**Reply by nurul3101** [reply]
@michaltaberski I am checking to see if you had a chance to look at my last response.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by michaltaberski** [reply]
Ok. Sorry. I will mark it as answered. I was using `include` already, but I was wondering if that's an expensive operation or not. The performance of the requests is fairly unknown to me ATM.
